instance/management/commands/migrate_swift_to_s3.py

```diff
@@ -0,0 +1,135 @@
+        command = "rclone copy ocim-swift:%s/%s ocim-s3:%s/%s" % (swift_container, key_name, s3_bucket, new_name)
+        print("Would run this command: %s.       ← FIXME test it and run it" % command)
+        return
+        subprocess.Popen(command, shell=True)
+
+    def _migrate_swift_to_s3(self, instance):
+        # FIXME Will these 2 values already be stored in the instance? If not, where to get them from? (Or should we set them and save?)
+        # FIXME and in particular: will the S3 region match the SWIFT region? (some clients don't want their data in a particular region; see comments on ticket)
+        self._create_rclone_config(instance.s3_access_key, instance.s3_secret_access_key)
+
+        copied = []  # FIXME describe variable
+        swift = self._get_swift_connection(instance)
+
+        container = instance.swift_container_name
+        s3_bucket = instance.s3_bucket_name  # FIXME will this instance already have the right bucket name? If not, we must set it and save the instance
+        LOG.info("Migrating container: %s" % container)
+        for item in swift.get_container(container)[1]:
+            LOG.info("Seeing file: %s", item['name'])
+
+            # FIXME test this part; will it also copy files in the root? (in addition to directories):
+            # get only directories
+            key_name = re.search('(.*\/)', item['name']).group(1)
+            if key_name in copied:
+                continue
+
+            # replace new name if they contain submissions_attachments (there is a bug on swift implementation that makes
+            # files be uploaded to the wrong directory)
+            # The changes to do are like:
+            #   submissions_attachmentsbadges/ ---> submissions_attachments/badges/
+            #   submissions_attachmentsuser_tasks ---> submissions_attachments/user_tasks/
+            # etc.
+            # FIXME what about submissions_attachmentssubmissions_attachments? In OC-4395 we needed: rclone-sync someserver/submissions_attachmentssubmissions_attachments/ someserver/submissions_attachments/
+            # FIXME test this part with more cases
+            new_name = re.sub(r'^(submissions_attachments)/?', '\g<1>/', key_name)
+            self._copy_subdirectory(key_name, new_name, swift_container=container, s3_bucket=s3_bucket)
+
+            copied.append(key_name)
+
+        # FIXME delete the rclone config to leave it empty for the next instance to migrate
+
+        LOG.info("Migrated! Copied: %i items. Full list: %s", len(copied), copied)
+
+    def handle(self, *args, **options):
+        """
+        Finds all instances and migrate them.
+        """
+        self.options = options
+
+        instances = OpenEdXInstance.objects.all()  # FIXME filter a little (e.g. exclude old/archived or those already in S3)
+
+        # Limit to a subset. Debugging only
+        instances = instances.filter(ref_set__name__contains='SWIFT') # FIXME disable
```

Comment:
we also have to filter by storage_type='swift'

Reply:
It would be nice to be able to pass a custom filter on the command line.  We may have different maintenance windows for different instances, since we need to coordinate the migration with clients.  If you want to get fancy, you can copy the logic for the `--filter` and `--exclude` parameters from the `instance_redeploy` management command, but it would be enough to accept a single instance id on the command line, and only migrate a single instance at a time.  We could then create the list of instance ids manually and use a shell loop to migrate multiple instances at once.  For paid instances, we may want to migrate them individually anyway, to make it easier to react to problems.